cmd/kube-scheduler/app/options/options.go

```diff
@@ -0,0 +1,234 @@
+	scheme.Default(&cfgv1alpha1)
+	cfg := componentconfig.KubeSchedulerConfiguration{}
+	if err := scheme.Convert(&cfgv1alpha1, &cfg, nil); err != nil {
+		return nil, err
+	}
+	return &cfg, nil
+}
+
+// AddFlags adds flags for a specific SchedulerServer to the specified FlagSet
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.ConfigFile, "config", o.ConfigFile, "The path to the configuration file. Flags override values in this file.")
+	fs.StringVar(&o.Master, "master", o.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+
+	o.InsecureServing.AddFlags(fs)
+	o.Deprecated.AddFlags(fs, &o.ComponentConfig)
+
+	leaderelectionconfig.BindFlags(&o.ComponentConfig.LeaderElection.LeaderElectionConfiguration, fs)
+	utilfeature.DefaultFeatureGate.AddFlag(fs)
+}
+
+func (o *Options) ApplyTo(c *schedulerconfig.Config) error {
+	if len(o.ConfigFile) == 0 {
+		glog.Warning("WARNING: all flags other than --config are deprecated. Please begin using a config file ASAP.")
+
+		if o.InsecureServing == nil {
+			o.ComponentConfig.HealthzBindAddress = ""
+		} else {
+			// override ComponentConfig.HealthzBindAddress with InsecureServing flag value if given.
+			if err := o.applyDeprecatedHealthzAddressToConfig(); err != nil {
+				return err
+			}
+			if err := o.applyDeprecatedHealthzPortToConfig(); err != nil {
+				return err
+			}
+
+			// sync back ComponentConfig.HealthzBindAddress to o.InsecureServing
+			if len(o.ComponentConfig.HealthzBindAddress) == 0 {
+				o.InsecureServing.BindPort = 0
+			} else {
+				host, port, err := net.SplitHostPort(o.ComponentConfig.HealthzBindAddress)
+				if err != nil {
+					return fmt.Errorf("invalid healthz bind address %q", o.ComponentConfig.HealthzBindAddress)
+				}
+				portInt, err := strconv.Atoi(port)
+				if err != nil {
+					return fmt.Errorf("invalid non-numerical healthz port %q", port)
+				}
+				o.InsecureServing.BindAddress = net.ParseIP(host)
+				o.InsecureServing.BindPort = portInt
+			}
+		}
+		if o.InsecureServing == nil && o.InsecureMetricsServing == nil {
+			o.ComponentConfig.MetricsBindAddress = ""
+		} else if o.InsecureServing != nil && len(o.ComponentConfig.MetricsBindAddress) > 0 && o.ComponentConfig.HealthzBindAddress != o.ComponentConfig.MetricsBindAddress {
+			// start a second http server for metrics
+			metricsServing := o.InsecureMetricsServing
+			if metricsServing == nil {
+				metricsServing = &controlleroptions.InsecureServingOptions{
+					BindNetwork: "tcp",
+				}
+			}
+			host, port, err := net.SplitHostPort(o.ComponentConfig.MetricsBindAddress)
+			if err != nil {
+				return fmt.Errorf("invalid metrics bind address %q", o.ComponentConfig.MetricsBindAddress)
+			}
+			portInt, err := strconv.Atoi(port)
+			if err != nil {
+				return fmt.Errorf("invalid non-numerical metrics port %q", port)
+			}
+			metricsServing.BindAddress = net.ParseIP(host)
+			metricsServing.BindPort = portInt
```

Comment:
if metricsServing and healthServing have same port but different BindAddress , will cause a conflict?